### DISCUSSION TITLE
Distinguish package version from flow list version

### DISCUSSION THREAD

**Discussion by bl-young** [USEPA/Federal-LCA-Commons-Elementary-Flow-List#205] [General]
Currently the package version and the flow list version are the same. This can cause confusion when a new version of the package is released but does not involve any changes to the flow list (e.g., new mapping files, new features, etc.).

I propose that we make a clean break such that the version of the flowlist does not have to align with the version of the fedelemflowlist package. However, the flowlist would indicate which version of fedelemflowlist created it.

I'm open to suggestions on where or how to make this transition.

**Comment by a-w-beck** [comment]
Agreed! The `fedelemflowlist` version number should not be identical to the [flowlist](https://www.lcacommons.gov/lca-collaboration/Federal_LCA_Commons/elementary_flow_list/datasets) data package version. An incrementation of the former _may_ lead to incrementing the latter, but not always. Furthermore, for each new flowlist release, the full `fedelemflowlist` software/model version number can be stored in the "Source information" repo metadata field. 

These patterns would also mirror how the USEEIO v2.0 [data package](https://www.lcacommons.gov/lca-collaboration/US_Environmental_Protection_Agency/USEEIO_v2/datasets) references "the precise USEEIO model version" from which it was generated without requiring the version numbers be identical.

**Reply by bl-young** [reply]
Thoughts on how to proceed at this particular point in time? Where we have a v1.2.3 released, though the most recent update to the flow list itself was back in v1.1.0. We will have new flows in the next release of fedelemflowlist (v1.2.4).

Perhaps the next release of the Flow List should be v1.3. (We have a few data objects out there that conflate both so I'm hesitant to use v1.2). And then we can increment the minor version on any changes to the flows, while patch versions can indicate metadata updates.

**Comment by bl-young** [comment]
Implemented a new spec `package_version` in https://github.com/USEPA/fedelemflowlist/commit/d35486e0da1e115b475e7adee6f89d414761f131. Most of the references to the version that exist will remain based on the List version (not the package version), including the output file naming.

**Comment by bnjmnmorelli** [comment]
The proposed change makes sense to me. Can you remind me of the formal name for the package relative to the flow list itself. If the names are easily distinguishable, confusion over version number seems less likely. Based on above posts, it seems like the updated name for the flow list itself would be fedelemflowlist v1.3? What would shorthand for the package be? Based on README text it seems like the package itself is named the same thing... @bl-young

**Reply by bl-young** [reply]
Correct, to date they have been the same thing, and IMO has led to confusion when people aren't sure if they are using the most up to date list (e.g., v1.1 is the last change that led to changes in the flowlist, despite the package being at v1.2.4). So by making those numbers too separate things I hope to reduce confusion.
